### ISSUE TITLE
Invizible pro DNScrypt don't work on mobile

### ISSUE THREAD

**Issue by itsbrody** [Gedsh/InviZible#209]
Just recently I stumbled over that app, finding it nice :-)
So installed from FDroid. As knowing TOR functionallity started it and DNScrypt at HOME (connected with WLAN) and all worked.
When leaving home (and my WLAN) I stumbled over an app problem of my mail app, not finding anymore its servers, so focusing on DNSCRYPT now and found that this piece does not start, when only mobile is available.

Its (repproducable) saying in the tab:
dnscrypt-proxy 2.1.4
Network not available yet -- waiting

When additivie switching WLAN, DNScrypt gets initialized immediately. Any suggestions?

**Comment by Gedsh** [comment]
> focusing on DNSCRYPT now and found that this piece does not start, when only mobile is available

Try checking if you are on an ipv6-only network. You can find it and try switching to ipv4 in your phone's APN settings.

Also, if you're really on an ipv6-only network, you can try the latest beta. It has full ipv6 support.

To use DNSCrypt on an ipv6-only network, you need to disable block_ipv6 and enable ipv6_servers in the DNSCrypt settings. Then select ipv6 DNSCrypt servers in Fast Settings.

**Comment by itsbrody** [comment]
Dear Oleksandr,

Thanks for your detailed advice.

Does your final sentence relate to the new beta only or could I check/change these settings also in my installed version with success?

Am 7. April 2023 19:28:49 MESZ schrieb Oleksandr Garmatin ***@***.***>:

**Comment by Gedsh** [comment]
You should use the latest beta version 1.8.3. Full ipv6 support is a new feature.

**Comment by itsbrody** [comment]
Just checked with beta version with same result, although hopefully switching on all you described and is needed. See screenshots

Exactly when adding proper WLAN , DnScrypt gets initialized as before.with released version. Please advice...

07.04.2023 21:51:52 Oleksandr Garmatin ***@***.***>:

**Comment by Gedsh** [comment]
Check the APN settings of your cellular network connection. It may be the wrong access point.

> Network not available yet -- waiting

This means you don't have a network connection at all. And neither InviZible nor DNSCrypt can see it. So you should check the network settings.

**Comment by itsbrody** [comment]
Will do (), but switching off Invisible Pro's DNSCrypt give "network connectivity" back just in that moment.
And any network using APP immediately starts to recover.

What aspects/attributes should I check, then?

Am 9. April 2023 16:56:18 MESZ schrieb Oleksandr Garmatin ***@***.***>:

**Comment by Gedsh** [comment]
> What aspects/attributes should I check, then?

APN and APN protocol.

**Comment by itsbrody** [comment]
thanks.
Because don't seeing any not normal, attaching the screenshots of my APN settings currently

Hth.

**Comment by Gedsh** [comment]
> attaching the screenshots

I don't see any attachments.

**Comment by Gedsh** [comment]
I see you are using APN internet.v6.telecom. V6 usually means that this access point only supports ipv6 regardless of other settings. You can try to look for another APN that supports ipv4.

Also try again using the latest beta 1.8.3 and follow the above recommendations to configure DNSCrypt to use ipv6. Uninstall the previous version and do a fresh install before trying. 

You can also try using [2620:fe::fe]:53 as DNSCrypt netprobe_address. To do this, you need to go to Menu -> DNSCrypt Settings and press Edit dnscrypt-proxy.toml directly.

If still no luck, please send me the logs. Menu -> Logs -> Collect logs.

**Comment by itsbrody** [comment]
> I see you are using APN internet.v6.telecom. V6 usually means that this access point only supports ipv6 regardless of other settings. You can try to look for another APN that supports ipv4.

I've checked against what Deutsche Telekom says @ (https://www.telekom.de/hilfe/mobilfunk/mobiles-internet/mobilfunkstandards/apn-ratgeber): Its the (accessable) name of their standard APN for modern tablets and mobiles - guessing a Fairphone FP3 as a modern android phone - currently serving under Android11. And as I could see - without using Invizable all other mobile capable APPs were and are using the mobile connectivity without any problems.
Might it be a problem for your system near APP that its a Multi-SIM phone, which is still not standard, where only one SIM is used currently?
 
The other tasks I will do later.

**Comment by Gedsh** [comment]
> standard APN for modern tablets and mobiles

Modern is good. But most likely you have a problem with ipv6.

> without using Invizable all other mobile capable APPs were and are using the mobile connectivity without any problems

InviZible is a special app designed for maximum privacy and anonymity. Its design is not so universal, therefore it has its drawbacks.

> Might it be a problem for your system near APP that its a Multi-SIM phone, which is still not standard, where only one SIM is used currently?

I don't think so.

**Comment by itsbrody** [comment]
I'm aware that InviZible is a special app and I like security.

 

For that I'm setting up step-by-step, also I know (and use) Tor solely on other devices, I decided as a first step to use DNScrypt.

Because on WLAN (an not very modern WLAN-Router) its working very well and altough having a modern phone, while should it have problems with IPV6 on mobile, but not on WLAN. And further this entry point (i.e. APN) has both capabilities and is setup adequately as Telecom says on its site. That does not make any sense for me....

 

Could we - with less effort on my side - check, that the multiSIM capability with only one SIM configured (on Android 11 currently) with your APP does not have this consequences, because my time to deeply test the APP is very limitted.
In the meantime I didn't find any spare time to go along your hints - thus I've disabled the APP currently completely.

 


 

Gesendet: Mittwoch, 19. April 2023 um 16:48 Uhr
Von: "Oleksandr Garmatin" ***@***.***>
An: "Gedsh/InviZible" ***@***.***>
Cc: "Wolfgang" ***@***.***>, "Author" ***@***.***>
Betreff: Re: [Gedsh/InviZible] Invizible pro DNScrypt don't work on mobile (Issue #209)


 


standard APN for modern tablets and mobiles


Modern is good. But most likely you have a problem with ipv6.


without using Invizable all other mobile capable APPs were and are using the mobile connectivity without any problems


InviZible is a special app designed for maximum privacy and anonymity. Its design is not so universal, therefore it has its drawbacks.


Might it be a problem for your system near APP that its a Multi-SIM phone, which is still not standard, where only one SIM is used currently?


I don't think so.
